src/python/pants/backend/python/dependency_inference/parse_python_dependencies.py

```diff
@@ -31,21 +34,44 @@ class ParsedPythonImports(FrozenDict[str, ParsedPythonImportInfo]):
     """All the discovered imports from a Python source file mapped to the relevant info."""
 
 
+class ParsedPythonAssets(DeduplicatedCollection[Tuple[str, str]]):
+    """All the discovered possible assets from a Python source file.
+
+    The tuple is of (containing module, filename).
+    """
+
+    # N.B. Don't set `sort_input`, as the input is already sorted
+
+
+@dataclass(frozen=True)
+class ParsedPythonDependencies:
+    imports: ParsedPythonImports
+    assets: ParsedPythonAssets
+
+
 @dataclass(frozen=True)
-class ParsePythonImportsRequest:
+class ParsePythonDependenciesRequest:
     source: PythonSourceField
     interpreter_constraints: InterpreterConstraints
     string_imports: bool
     string_imports_min_dots: int
+    assets: bool
+    asset_min_slashes: int
+
+
+def _filepath_to_modname(filepath: str) -> str:
+    return str(pathlib.Path(filepath).with_suffix("")).replace("/", ".")
```

Comment:
We try to limit ourselves to `PurePath` to avoid the risk of using `Path`'s IO side effects.